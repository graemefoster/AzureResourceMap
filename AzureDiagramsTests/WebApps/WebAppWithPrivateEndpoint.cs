using AzureDiagrams.Resources;
using AzureDiagrams.Resources.Retrievers.Extensions;
using Shouldly;

namespace AzureDiagramsTests.WebApps;

public class WebAppWithPrivateEndpoint
{
    [Fact]
    public async Task CanDrawDiagram()
    {
        var resources = BuildScenario();

        var diagram = await AzureDiagramGenerator.DrawIoDiagramGenerator.DrawDiagram(
            resources,
            false,
            false,
            false,
            true,
            false);

        diagram.ShouldMatchApproved();
    }

    
    [Fact]
    public async Task CanDrawCondensedDiagram()
    {
        var resources = BuildScenario();

        var diagram = await AzureDiagramGenerator.DrawIoDiagramGenerator.DrawDiagram(
            resources,
            true,
            false,
            false,
            true,
            false);

        diagram.ShouldMatchApproved();
    }

    private static AzureResource[] BuildScenario()
    {
        var vnetId = "/subscriptions/123/resourceGroups/rg1/providers/microsoft.network/virtualnetworks/vnet123";

        var vnet = new VNet(vnetId, "test-vnet", new[]
        {
            new VNet.Subnet("private-endpoints", "10.1.1.0/24"),
            new VNet.Subnet("vnet-integration", "10.1.2.0/24"),
        });

        var appServicePlanId =
            "/subscriptions/123/resourceGroups/rg1/providers/Microsoft.Web/serverFarms/TestAppServicePlan";

        var asp = new AppServicePlan(appServicePlanId, "TestAppServicePlan");

        var peId = new Guid("36a566f9-f6b8-439c-9371-204842160e2b").ToString();
        var pe2Id = new Guid("36a566f9-f6b8-439c-9371-204842160e2d").ToString();

        var app1HostName = "TestApp1.azurewebsites.net";
        var app2HostName = "TestApp2.azurewebsites.net";

        var app1 = new AppServiceApp("/subscriptions/123/resourceGroups/rg1/providers/Microsoft.Web/sites/TestApp1",
            appServicePlanId,
            "TestApp1",
            false,
            Array.Empty<string>(),
            new[] { app1HostName },
            virtualNetworkSubnetId: $"{vnetId}/subnets/vnet-integration",
            new PrivateEndpointExtensions(new[] { peId }));

        var app2 = new AppServiceApp("/subscriptions/123/resourceGroups/rg1/providers/Microsoft.Web/sites/TestApp2",
            appServicePlanId,
            "TestApp2",
            false,
            new[] { $"https://{app1HostName}" },
            new[] { app2HostName },
            virtualNetworkSubnetId: $"{vnetId}/subnets/vnet-integration",
            new PrivateEndpointExtensions(new[] { pe2Id }));

        var nicId = new Guid("36a566f9-f6b8-439c-9371-204842160e2c").ToString();
        var nic = new Nic(nicId,
            IpConfigurations.ForPrivateEndpoint("10.1.1.1", $"{vnetId}/subnets/private-endpoints", app1HostName));

        var privateEndpoint = new PrivateEndpoint(peId, new[] { $"{vnetId}/subnets/private-endpoints" },
            new[] { nic.Id }, new[] { $"https://{app1HostName}" });

        var nic2Id = new Guid("36a566f9-f6b8-439c-9371-204842160e2e").ToString();
        var nic2 = new Nic(nic2Id,
            IpConfigurations.ForPrivateEndpoint("10.1.1.2", $"{vnetId}/subnets/private-endpoints", app2HostName));

        var privateEndpoint2 = new PrivateEndpoint(pe2Id, new[] { $"{vnetId}/subnets/private-endpoints" },
            new[] { nic2.Id }, new[] { $"https://{app2HostName}" });

        var azureResources = new AzureResource[]
        {
            vnet,
            asp,
            app1,
            app2,
            nic,
            nic2,
            privateEndpoint,
            privateEndpoint2
        };

        var resources = azureResources.Process();
        return resources;
    }
}