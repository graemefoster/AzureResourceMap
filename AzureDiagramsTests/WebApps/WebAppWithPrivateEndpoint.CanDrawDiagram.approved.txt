<mxGraphModel>
	<root>
		<mxCell id="0" />
		<mxCell id="1" parent="0" />
<mxCell id="66f228a5-2df7-c23b-4814-95adb050732d" value="" style="html=1;image;image=img/lib/azure2/networking/Network_Interfaces.svg;fontSize=12;labelPosition=bottom" vertex="1" parent="bd3a1a44-a441-608a-7d61-14ee6b5f190c.private-endpoints">
    <mxGeometry x="50" y="125" width="50" height="50" 
    as="geometry" />
</mxCell>
<mxCell id="f1921020-766f-e0b6-735f-33e982981b99" value="" style="html=1;image;image=img/lib/azure2/networking/Network_Interfaces.svg;fontSize=12;labelPosition=bottom" vertex="1" parent="bd3a1a44-a441-608a-7d61-14ee6b5f190c.private-endpoints">
    <mxGeometry x="200" y="125" width="50" height="50" 
    as="geometry" />
</mxCell>
<mxCell id="0fdd30e6-bd02-cb30-415c-a4b74d440c1b" value="" style="html=1;image;image=img/lib/azure2/networking/Private_Link.svg;fontSize=12;labelPosition=bottom" vertex="1" parent="bd3a1a44-a441-608a-7d61-14ee6b5f190c.private-endpoints">
    <mxGeometry x="50" y="50" width="50" height="50" 
    as="geometry" />
</mxCell>
<mxCell id="367449a7-6bdf-3242-7a62-e8447542c82f" value="" style="html=1;image;image=img/lib/azure2/networking/Private_Link.svg;fontSize=12;labelPosition=bottom" vertex="1" parent="bd3a1a44-a441-608a-7d61-14ee6b5f190c.private-endpoints">
    <mxGeometry x="200" y="50" width="50" height="50" 
    as="geometry" />
</mxCell>
<mxCell id="05f10308-a191-af78-fcb8-3e9e99244b55" value="TestApp1" style="html=1;image;image=img/lib/azure2/networking/Virtual_Networks.svg;fontSize=12;labelPosition=bottom" vertex="1" parent="bd3a1a44-a441-608a-7d61-14ee6b5f190c.vnet-integration">
    <mxGeometry x="50" y="50" width="50" height="50" 
    as="geometry" />
</mxCell>
<mxCell id="8508a9a2-8be4-6d67-545a-bfa01a44cc12" value="TestApp2" style="html=1;image;image=img/lib/azure2/networking/Virtual_Networks.svg;fontSize=12;labelPosition=bottom" vertex="1" parent="bd3a1a44-a441-608a-7d61-14ee6b5f190c.vnet-integration">
    <mxGeometry x="200" y="50" width="50" height="50" 
    as="geometry" />
</mxCell>
<mxCell id="5edd63b5-ec09-fe25-5a63-f2f6dadbc0bd" value="TestAppServicePlan" style="html=1;image;image=img/lib/azure2/app_services/App_Service_Plans.svg;fontSize=12;labelPosition=bottom" vertex="1" parent="5edd63b5-ec09-fe25-5a63-f2f6dadbc0bd.container">
    <mxGeometry x="350" y="50" width="50" height="50" 
    as="geometry" />
</mxCell>
<mxCell id="60c80ab3-baea-d6d4-b81c-54b7d72ac690" value="TestApp1" style="html=1;image;image=img/lib/azure2/app_services/App_Services.svg;fontSize=12;labelPosition=bottom" vertex="1" parent="5edd63b5-ec09-fe25-5a63-f2f6dadbc0bd.container">
    <mxGeometry x="50" y="50" width="50" height="50" 
    as="geometry" />
</mxCell>
<mxCell id="d18af89b-56f1-32d3-2249-8fc306f5363b" value="TestApp2" style="html=1;image;image=img/lib/azure2/app_services/App_Services.svg;fontSize=12;labelPosition=bottom" vertex="1" parent="5edd63b5-ec09-fe25-5a63-f2f6dadbc0bd.container">
    <mxGeometry x="200" y="50" width="50" height="50" 
    as="geometry" />
</mxCell>
<mxCell id="bd3a1a44-a441-608a-7d61-14ee6b5f190c.private-endpoints" value="private-endpoints&lt;br/&gt;(10.1.1.0/24)" style="rounded=0;whiteSpace=wrap;html=1;fillColor=white;verticalAlign=top" vertex="1" parent="bd3a1a44-a441-608a-7d61-14ee6b5f190c">
    <mxGeometry x="50" y="50" width="300" height="235" 
    as="geometry" />
</mxCell>
<mxCell id="bd3a1a44-a441-608a-7d61-14ee6b5f190c.vnet-integration" value="vnet-integration&lt;br/&gt;(10.1.2.0/24)" style="rounded=0;whiteSpace=wrap;html=1;fillColor=white;verticalAlign=top" vertex="1" parent="bd3a1a44-a441-608a-7d61-14ee6b5f190c">
    <mxGeometry x="50" y="310" width="300" height="160" 
    as="geometry" />
</mxCell>
<mxCell id="bd3a1a44-a441-608a-7d61-14ee6b5f190c" value="test-vnet&lt;br/&gt;(VNet)" style="rounded=0;whiteSpace=wrap;html=1;fillColor=#F8CECC;verticalAlign=top" vertex="1" parent="207c67fc-34bc-d9f0-369e-0f16975fc114.container">
    <mxGeometry x="75" y="236.6875" width="400" height="530" 
    as="geometry" />
</mxCell>
<mxCell id="bd3a1a44-a441-608a-7d61-14ee6b5f190c.image.0" style="html=1;image;image=img/lib/azure2/networking/Virtual_Networks.svg;fontSize=12;labelPosition=bottom" vertex="1" parent="bd3a1a44-a441-608a-7d61-14ee6b5f190c">
    <mxGeometry x="360" y="490" width="30" height="30" 
    as="geometry" />
</mxCell>
<mxCell id="5edd63b5-ec09-fe25-5a63-f2f6dadbc0bd.container" value="TestAppServicePlan&lt;br/&gt;(microsoft.web/serverfarms)" style="rounded=0;whiteSpace=wrap;html=1;fillColor=#DAE8FC;verticalAlign=top" vertex="1" parent="207c67fc-34bc-d9f0-369e-0f16975fc114.container">
    <mxGeometry x="50" y="50" width="450" height="160" 
    as="geometry" />
</mxCell>
<mxCell id="207c67fc-34bc-d9f0-369e-0f16975fc114.container" value="global&lt;br/&gt;(Region)" style="rounded=0;whiteSpace=wrap;html=1;fillColor=#F5F5F5;verticalAlign=top" vertex="1" parent="1">
    <mxGeometry x="0" y="0" width="550" height="826.6875" 
    as="geometry" />
</mxCell>
<mxCell id="ddb3db1b-e793-b5a7-3215-7e30a151af16" 
         style="edgeStyle=orthogonalEdgeStyle;rounded=0;orthogonalLoop=1;jettySize=auto;html=1;" edge="1" parent="1" 
         source="60c80ab3-baea-d6d4-b81c-54b7d72ac690" target="05f10308-a191-af78-fcb8-3e9e99244b55">
            <mxGeometry relative="1" as="geometry" />
             </mxCell>
             
<mxCell id="6fd81717-3d3d-2205-7183-0b2ed552f022" 
         style="edgeStyle=orthogonalEdgeStyle;rounded=0;orthogonalLoop=1;jettySize=auto;html=1;" edge="1" parent="1" 
         source="d18af89b-56f1-32d3-2249-8fc306f5363b" target="8508a9a2-8be4-6d67-545a-bfa01a44cc12">
            <mxGeometry relative="1" as="geometry" />
             </mxCell>
             
<mxCell id="31164425-6f47-4d2c-c6b1-28566c8fcfdf" 
         style="edgeStyle=orthogonalEdgeStyle;rounded=0;orthogonalLoop=1;jettySize=auto;html=1;" edge="1" parent="1" 
         source="66f228a5-2df7-c23b-4814-95adb050732d" target="0fdd30e6-bd02-cb30-415c-a4b74d440c1b">
            <mxGeometry relative="1" as="geometry" />
             </mxCell>
             
<mxCell id="37954a19-2370-4b45-9726-c907604a11a8" 
         style="edgeStyle=orthogonalEdgeStyle;rounded=0;orthogonalLoop=1;jettySize=auto;html=1;" edge="1" parent="1" 
         source="f1921020-766f-e0b6-735f-33e982981b99" target="367449a7-6bdf-3242-7a62-e8447542c82f">
            <mxGeometry relative="1" as="geometry" />
             </mxCell>
             
<mxCell id="03b7c2b8-0c1b-7a4a-c51d-0aa7117832a5" 
         style="edgeStyle=orthogonalEdgeStyle;rounded=0;orthogonalLoop=1;jettySize=auto;html=1;" edge="1" parent="1" 
         source="0fdd30e6-bd02-cb30-415c-a4b74d440c1b" target="60c80ab3-baea-d6d4-b81c-54b7d72ac690">
            <mxGeometry relative="1" as="geometry" />
             </mxCell>
             
<mxCell id="4bd35d0b-db6a-f474-1560-278b55832753" 
         style="edgeStyle=orthogonalEdgeStyle;rounded=0;orthogonalLoop=1;jettySize=auto;html=1;" edge="1" parent="1" 
         source="367449a7-6bdf-3242-7a62-e8447542c82f" target="d18af89b-56f1-32d3-2249-8fc306f5363b">
            <mxGeometry relative="1" as="geometry" />
             </mxCell>
             
<mxCell id="32917fb3-76cd-325d-0425-483011b23905" 
         style="edgeStyle=orthogonalEdgeStyle;rounded=0;orthogonalLoop=1;jettySize=auto;html=1;" edge="1" parent="1" 
         source="8508a9a2-8be4-6d67-545a-bfa01a44cc12" target="66f228a5-2df7-c23b-4814-95adb050732d">
            <mxGeometry relative="1" as="geometry" />
             </mxCell>
             <mxCell id="32917fb3-76cd-325d-0425-483011b23905--1" value="calls" style="edgeLabel;html=1;align=center;verticalAlign=middle;resizable=0;points=[];" vertex="1" connectable="0" parent="32917fb3-76cd-325d-0425-483011b23905">
             <mxGeometry x="-0.5" relative="1" as="geometry">
             <mxPoint as="offset" />
             </mxGeometry>
             </mxCell>

	</root>
</mxGraphModel>